internal/pkg/cli/secret_init.go

```diff
@@ -75,20 +98,97 @@ func (o *secretInitOpts) Validate() error {
 }
 
 func (o *secretInitOpts) Ask() error {
+	if o.overwrite {
+		log.Infof("You have specified %s flag. Please note that overwriting an existing secret may break your deployed service.\n", color.HighlightCode("--overwrite"))
+	}
+	if err := o.askForAppName(); err != nil {
+		return err
+	}
+	if err := o.askForSecretName(); err != nil {
+		return err
+	}
+	if err := o.askForSecretValues(); err != nil {
+		return err
+	}
 	return nil
 }
 
 func (o *secretInitOpts) Execute() error {
 	return nil
 }
 
+func (o *secretInitOpts) askForAppName() error {
+	if o.appName != "" {
+		return nil
+	}
+
+	app, err := o.selector.Application(secretInitAppPrompt, secretInitAppPromptHelp)
+	if err != nil {
+		return fmt.Errorf("ask for an application to add the secret to: %w", err)
+	}
+	o.appName = app
+	return nil
+}
+
+func (o *secretInitOpts) askForSecretName() error {
+	if o.name != "" {
+		return nil
+	}
+
+	name, err := o.prompter.Get(secretInitSecretNamePrompt,
+		secretInitSecretNamePromptHelp,
+		validateSecretName,
+		prompt.WithFinalMessage("secret name: "))
+	if err != nil {
+		return fmt.Errorf("ask for the secret name: %w", err)
+	}
+
+	o.name = name
+	return nil
+}
+
+func (o *secretInitOpts) askForSecretValues() error {
+	if o.values != nil {
+		return nil
+	}
+
+	envs, err := o.store.ListEnvironments(o.appName)
+	if err != nil {
+		return fmt.Errorf("list environments in app %s: %w", o.appName, err)
+	}
+
+	if len(envs) == 0 {
+		log.Errorf("Secrets are environment-level resource. Please run %s before running %s.\n",
+			color.HighlightCode("copilot env init"),
+			color.HighlightCode("copilot secret init"))
+		return fmt.Errorf("no environment is found in app %s", o.appName)
+	}
+
+	values := make(map[string]string)
+	for _, env := range envs {
+		value, err := o.prompter.GetSecret(
+			fmt.Sprintf(fmtSecretInitSecretValuePrompt, color.HighlightUserInput(o.name), env.Name),
+			fmt.Sprintf(fmtSecretInitSecretValuePromptHelp, color.HighlightUserInput(o.name), env.Name))
+		if err != nil {
+			return fmt.Errorf("ask for secret %s's value in environment %s: %w",
```

Comment:
The ask doesn't have any error. It should be get error instead.
```suggestion
			return fmt.Errorf("get secret value for %s in environment %s: %w",
```